### PR TITLE
Avoid device sync in training loss accumulation

### PR DESCRIPTION
**Pull request by cyyever**
# What does this PR do?
This PR avoids device sync in training loss accumulation by ```torch.where```. The `is_torch_xla_available` condition is also removed.

### CODE REVIEW

**Comment by Rocketknight1** [comment]
Hey @cyyever! We did some internal discussion about CUDA sync PRs - would you be able to make a benchmark or profile graph showing the reduction in syncs? We might make those mandatory in future just because it can sometimes be hard to tell from reading the code where exactly syncs would occur
